### PR TITLE
[BE][MPS] Prefer xfail to skip

### PR DESCRIPTION
**Pull request by malfet**
This essentially undoes large skips on everything but MacOS Sequoia to nn.modules made by https://github.com/pytorch/pytorch/pull/128393

Instead it uses existing `xfail`, but guards it on `_macos15_or_newer` boolean

Before the change if run on MacOS 14:
```
 % python3 ../test/test_modules.py -v -k Hardswish 2>&1|tail -n3
Ran 57 tests in 0.053s

OK (skipped=32)
```
After
```
% python3 ../test/test_modules.py -v -k Hardswish 2>&1|tail -n3
Ran 57 tests in 0.229s

OK (skipped=10, expected failures=2)
```

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
This PR assumes that the tests all pass on macos15 or newer, yes?
Failing CI is relevant.

**Comment by malfet** [comment]
@janeyx99 indeed they are, I haven't realized that `skipMPSVersionIfLessThan` crept into test_nn.py...

File: torch/testing/_internal/common_modules.py [hunk 1]
```diff
@@ -4273,20 +4288,28 @@ def module_error_inputs_torch_nn_Pad3d(module_info, device, dtype, requires_grad
...
+                       'TestModule',
+                       'test_memory_format',
+                       active_if=operator.itemgetter('training'),
+                       device_type='mps',
                    ),)
                ),
     ModuleInfo(torch.nn.Tanhshrink,
                module_inputs_func=module_inputs_torch_nn_Tanhshrink,
-               skips=(
+               skips=None if _macos15_or_newer else (
```
Comment (janeyx99) [inline torch/testing/_internal/common_modules.py:4304]: Do we also want the skips here to be None for non macos devices? like 
```suggestion
               skips=None if _macos15_or_newer or not is_macos else (
```
where is_macos is defined as being on macos at the time?

**Review by janeyx99 — APPROVED** [review]
One nitty comment

File: torch/testing/_internal/common_modules.py (hunk 1 above)
Comment (malfet) [reply torch/testing/_internal/common_modules.py:4304]: It's a bit confusing, but it works as expected as skip is only for `device_type='mps'`

Comment (janeyx99) [reply torch/testing/_internal/common_modules.py:4304]: Yea I saw that, I was more thinking about the unnecessary overhead

**Pull request closed without merging**